### ISSUE TITLE
[bug] - WalletConnect QR Code not working with Metamask

### ISSUE THREAD

**Issue by 70nyIT** [WalletConnect/web3modal#957]
### Link to minimal reproducible example

https://codesandbox.io/s/interesting-water-lr0i46?file=/src/index.tsx

### Summary

When attempting to connect to WalletConnect by scanning the QR code on my desktop with the Metamask app installed on my phone, the QR code is either not scanned or causes the app to crash. I have experienced this issue on both Android and iOS apps.

### Steps to Reproduce
1. Click the "Connect Wallet" button in the sandbox code above or at https://lr0i46.csb.app/.
2. Open the Metamask app on your smartphone and scan the QR code to connect using WalletConnect.

### Expected Behavior
Metamask should connect without any issues.

### Actual Behavior
After scanning the QR code with Metamask, the camera view closes, but nothing happens on either the mobile app or on the screen.

However, if I read the QR code with another wallet app, such as TrustWallet, it works with no problems. It is worth noting that using RainbowKit (https://github.com/rainbow-me/rainbowkit) I can connect to WalletConnect using Metamask.

This bug is important to point out due to the popularity of Metamask wallet.

### List of related npm package versions

```
{
  ...
  "dependencies": {
    ...
    "@web3modal/ethereum": "2.2.2",
    "@web3modal/react": "2.2.2",
    "ethers": "^5",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "wagmi": "^0.12.6"
    ...
  },
  ...
}
```

**Comment by 0xAsimetriq** [comment]
`version: 2` is not supported by metamask mobile app at the moment (it is in progress though). To use it with metamask you'll have to set `version: 1`

**Comment by 70nyIT** [comment]
Okay. 
I had experiences with Mobile Apps and it's incredible when a new release is delivered, how long takes to have 90%+ of your users to that version. I bet that if metamask integrates V2 in the next month, by the end of June not 90%+ of users will have updated the App. 

Question: would be possible on Web3Modal to show a message saying "you're trying to connect using v1, switch to v2"? Or something that shows it's a user-side issue?

TLDR: let's hope they'll integrate ASAP. As of now, I'll leave v1, and will migrate to V2 in June.
